macOS/DuckDuckGo/NetworkProtection/AppTargets/DeveloperIDTarget/VPNSubscriptionEventsHandler.swift

```diff
@@ -60,10 +62,24 @@ final class NetworkProtectionSubscriptionEventHandler {
     private func checkEntitlements() {
         Task {
             let hasEntitlement = await subscriptionManager.isFeatureEnabledForUser(feature: .networkProtection)
-            await handleEntitlementsChange(hasEntitlements: hasEntitlement, source: .clientCheck(sourceObject: self))
+            await handleEntitlementsChange(hasEntitlements: hasEntitlement, trigger: .clientCheck)
         }
     }
 
+    private func subscribeToWakeNotifications() {
+        NSWorkspace.shared.notificationCenter
+            .publisher(for: NSWorkspace.didWakeNotification)
+            .receive(on: DispatchQueue.main)
+            .sink { [weak self] _ in
+                Logger.networkProtection.log("System wake notification received, checking entitlements")
+                Task {
+                    let hasEntitlement = await self?.subscriptionManager.isFeatureEnabledForUser(feature: .networkProtection) ?? false
```

Comment:
This check differ from iOS where we check both subscription features and user entitlements; Here is just user entitlements, we should align them.
Options:
1. use `isFeatureEnabledForUser(feature` everywhere
2. use `isFeatureAvailableAndEnabled(feature` everywwhere